src/api/controllers/signup.controller.js

```diff
@@ -0,0 +1,62 @@
+import { v4 as uuidv4 } from "uuid";
+import bcrypt from "bcrypt";
+import { sendEmail } from "../utils/rondom-email";
+import { hashPassword } from "../utils/hash-password";
+import { signUpUser } from "../services/user.service";
+import ERoles from "../enums/ERole";
+import { StatusCodes } from "http-status-codes";
+
+export const signUpDriver = async (req, res) => {
+ 
+  const { name,email } = req.body;
+
+  // Generate a random password
+  const password = uuidv4().substr(0, 8);
+  console.log("password: " + password);
+
+  // Hash the password
+  const hashedPassword = await bcrypt.hash(password, saltRounds);
+
+  // Create the new driver
+  const newDriver = { name,email, password: hashedPassword, role: ERoles.DRIVER };
+
+  // Add the new driver to the list
+  const driverData = await signUpUser(newDriver);
+
+  // Send an email to the provider email with the random password
+  // await sendEmail(email, 'Your password for the app', `Your password is ${password}`);
+
+  // Return a success message
+  res.status(StatusCodes.CREATED).json({
+    success: true,
+    message: "Driver registered successfully",
+    data: driverData,
+  });
```

Comment:
create one controller that will receive 
- will receive that body from the postman `name,email,role(operator or driver)`
- find role By RoleTitle `const role=findOne(req.body.role) `to get role `id`
- `const newUser={name,email,password:hashedPassword,role_id:role.id}` 
- finally save newUser